src/components/LayerUI.tsx

```diff
@@ -164,69 +178,32 @@ const LibraryMenu = ({
   const ref = useRef<HTMLDivElement | null>(null);
   useOnClickOutside(ref, onClickOutside);
 
-  const [libraryItems, setLibraryItems] = useState<LibraryItems>([]);
-
-  const [loadingState, setIsLoading] = useState<
-    "preloading" | "loading" | "ready"
-  >("preloading");
-
-  const loadingTimerRef = useRef<NodeJS.Timeout | null>(null);
-
-  useEffect(() => {
-    Promise.race([
-      new Promise((resolve) => {
-        loadingTimerRef.current = setTimeout(() => {
-          resolve("loading");
-        }, 100);
-      }),
-      loadLibrary().then((items) => {
-        setLibraryItems(items);
-        setIsLoading("ready");
-      }),
-    ]).then((data) => {
-      if (data === "loading") {
-        setIsLoading("loading");
-      }
-    });
-    return () => {
-      clearTimeout(loadingTimerRef.current!);
-    };
-  }, []);
+  const libraryItems = useGlobalLibrary();
 
-  const removeFromLibrary = useCallback(async (indexToRemove) => {
-    const items = await loadLibrary();
-    const nextItems = items.filter((_, index) => index !== indexToRemove);
-    saveLibrary(nextItems);
-    setLibraryItems(nextItems);
+  const removeFromLibrary = useCallback((indexToRemove) => {
+    globalLibraryState.replaceLibrary((items) =>
+      items.filter((_, index) => index !== indexToRemove),
+    );
   }, []);
 
   const addToLibrary = useCallback(
-    async (elements: NonDeleted<ExcalidrawElement>[]) => {
-      const items = await loadLibrary();
-      const nextItems = [...items, elements];
+    (elements: NonDeleted<ExcalidrawElement>[]) => {
+      globalLibraryState.replaceLibrary((items) => [...items, elements]);
       onAddToLibrary();
-      saveLibrary(nextItems);
-      setLibraryItems(nextItems);
     },
     [onAddToLibrary],
   );
 
-  return loadingState === "preloading" ? null : (
+  return (
     <Island padding={1} ref={ref} className="layer-ui__library">
-      {loadingState === "loading" ? (
-        <div className="layer-ui__library-message">
-          {t("labels.libraryLoadingMessage")}
-        </div>
-      ) : (
```

Comment:
Why remove the loading state? Maybe it was a bit far fetched for an IDB case, I admit, but would still be useful when we support persisting to server.